lib/src/store.dart

```diff
@@ -83,26 +102,40 @@ class Store {
   /// If [onAction] is provided, it will be called every time [action] is
   /// dispatched. If [onAction] returns a [Future], [trigger] will not be
   /// called until that future has resolved.
+  ///
+  /// If the `Store` has been disposed, this method throws a [StateError].
   triggerOnAction(Action action, [void onAction(payload)]) {
+    if (isDisposed) {
+      throw new StateError('Store has been disposed');
+    }
+
     if (onAction != null) {
-      action.listen((payload) async {
+      _actionSubscriptions.add(action.listen((payload) async {
         await onAction(payload);
         trigger();
-      });
+      }));
     } else {
-      action.listen((_) {
+      _actionSubscriptions.add(action.listen((_) {
         trigger();
-      });
+      }));
     }
   }
 
   /// Adds a subscription to this `Store`.
   ///
   /// Each time this `Store` triggers (by calling [trigger]), indicating that
   /// data has been mutated, [onData] will be called.
+  ///
+  /// If the `Store` has been disposed, this method throws a [StateError].
   StreamSubscription<Store> listen(StoreHandler onData,
       {Function onError, void onDone(), bool cancelOnError}) {
-    return _stream.listen(onData,
+    if (isDisposed) {
+      throw new StateError('Store has been disposed');
+    }
+
+    StreamSubscription<Store> sub = _stream.listen(onData,
         onError: onError, onDone: onDone, cancelOnError: cancelOnError);
+    manageStreamSubscription(sub);
```

Comment:
Hmm, yeah, but we never clean that list of subscriptions up, so as components subscribe and then dispose themselves we'll end up with stale (canceled) subscriptions lying around.